lib/Frontend/CompilerInvocation.cpp

```diff
@@ -166,36 +134,57 @@ class ArgsToFrontendInputsConverter {
   bool convert() {
     if (enforceFilelistExclusion())
       return true;
-    getFilesFromCommandLine();
-    if (getFilesFromFilelist())
+    if (getFilesFromCommandLine() || getFilesFromFilelist())
```

Comment:
This is really subtle for me, and seems to conflate two things: "we were able to load" and "we had an error". I don't think not having files on the command line should be considered an error, or at least not the same kind of error as what comes from `getFilesFromFilelist`. (If we made errors explicit instead of just `true` this would be a lot more obvious.)

I don't have something *better* at the moment, but maybe you do. Or maybe we should just go ahead.

Reply:
I'll rewrite so it is not so subtle. At present, getFilesFromCommandLine always returns false, succeeded, because I elected to not bail on duplicates at this stage--too many tests fail.
There's a comment to that effect down below.